workers/unity/Packages/com.improbable.gdk.core/Worker/Worker.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using Improbable.Worker;
+using Improbable.Worker.Core;
+using Unity.Entities;
+using UnityEngine;
+using Entity = Unity.Entities.Entity;
+
+namespace Improbable.Gdk.Core
+{
+    public class Worker : IDisposable
+    {
+        public const long WorkerEntityId = -1337;
+
+        internal readonly Dictionary<EntityId, Entity> EntityMapping = new Dictionary<EntityId, Entity>();
+
+        public readonly Vector3 Origin;
+        public readonly string WorkerType;
+        public readonly string WorkerId;
+        public readonly World World;
+        public readonly Connection Connection;
+        public readonly ILogDispatcher LogDispatcher;
+
+        public delegate void WorkerEventHandler(Worker worker);
+        public static event WorkerEventHandler OnConnect;
+        public static event WorkerEventHandler OnDisconnect;
+
+        private Worker(ConnectionConfig config, ILogDispatcher logDispatcher, Vector3 origin)
+        {
+            Origin = origin;
+            WorkerId = string.IsNullOrEmpty(config.WorkerId)
+                ? $"{config.WorkerType}-{Guid.NewGuid()}"
+                : config.WorkerId;
+
+            Connection = ConnectionUtility.Connect(config, WorkerId);
+            LogDispatcher = logDispatcher;
+            WorkerType = config.WorkerType;
+            if (LogDispatcher is ForwardingDispatcher dispatcher)
+            {
+                dispatcher.SetConnection(Connection);
+            }
+
+            World = new World(WorkerId);
+            var workerSystem = World.GetOrCreateManager<WorkerSystem>();
+            workerSystem.Worker = this;
+
+            var entityManager = World.GetOrCreateManager<EntityManager>();
+            var entity = entityManager.CreateEntity();
```

Comment:
Please pass in the ComponentTypes before hand (or an archetype) and then set the components.
Otherwise we get the chunk allocation again magic.

Reply:
Actually.. since those components dont have any data, just passing in the types should do the trick